rasa/core/policies/ted_policy.py

```diff
@@ -481,10 +745,11 @@ def load(cls, path: Union[Text, Path]) -> "TEDPolicy":
             model_data_example,
             data_signature=model_data_example.get_signature(),
             config=meta,
-            max_history_tracker_featurizer_used=isinstance(
-                featurizer, MaxHistoryTrackerFeaturizer
-            ),
+            # during prediction we don't care about previous dialogue turns,
+            # so to save computation time, use only the last one
```

Comment:
I think it's worth mentioning here that each dialogue turn would have embedded the dialogue from start, and hence using the embedding from just the last turn is fine.
I was confused why would it be so for a minute because the comment/variable name gives the impression that the embedding of just the last event in the tracker is needed to make the prediction.

Reply:
Wait now I am confused, when I look at `_embed_dialogue`, it looks like this variable is being used to basically take the last token representation, a.k.a. CLS token representation. This would be done during training also, no?

Reply:
@Ghostvv could you please take this over

Reply:
> take the last token representation, a.k.a. CLS token representation.

yes but not CLS, here the last token has different meaning. It is the last dialogue turn

> This would be done during training also, no?

it depends on the featurizer. For max history one - yes, for full dialogue - no. However, during prediction we don't care for previous dialogue turn predictions regardless of the featurizer

Reply:
Right that makes sense! 👍